plugin/pkg/admission/webhook/admission.go

```diff
@@ -84,15 +77,32 @@ type WebhookSource interface {
 }
 
 // NewGenericAdmissionWebhook returns a generic admission webhook plugin.
-func NewGenericAdmissionWebhook() (*GenericAdmissionWebhook, error) {
+func NewGenericAdmissionWebhook(configFile io.Reader) (*GenericAdmissionWebhook, error) {
+	kubeconfigFile := ""
+	if configFile != nil {
+		// TODO: move this to a versioned configuration file format
+		var config AdmissionConfig
+		d := yaml.NewYAMLOrJSONDecoder(configFile, 4096)
+		err := d.Decode(&config)
+		if err != nil {
+			return nil, err
+		}
+		kubeconfigFile = config.AdmissionWebhookConfig.KubeConfigFile
```

Comment:
shouldn't need the extra layer of nesting... I thought the `configFile` reader you got here was just the config for this admission plugin

Reply:
> shouldn't need the extra layer of nesting... I thought the configFile reader you got here was just the config for this admission plugin

I followed the pattern from the ImagePolicyWebhook.  I agree and I'll fix up the TODO about creating a versioned type, but adding types drowns out content.  How about we make ourselves consistent with other webhooks and I promise to create a versioned config in the next 24 hours?

Reply:
> shouldn't need the extra layer of nesting... I thought the configFile reader you got here was just the config for this admission plugin

I'll remove the layering here and do versioned types in a close followup.